wordpress-mobile/gutenberg-mobile
# Publish React Native Bridge to Bintray from CI

- [ ] Before merging this PR, we need to merge https://github.com/WordPress/gutenberg/pull/29488 into Gutenberg's `trunk` and update the submodule has here.
- [x] Before merging this PR, it needs to implement the versioning strategy specified [here](https://github.com/wordpress-mobile/gutenberg-mobile/pull/3262/files#diff-11096f9d861fef21abf4673f7b86333f1d5dfb14018b31593c81d88f6f00f446R37-R43)

---

This PR adds CI jobs to build and publish the AAR for the `react-native-bridge` package (which lives in Gutenberg).

I originally planned to open this against the `gutenberg-composite-build` project WIP branch, but given this doesn't change anything in the codebase itself other than the Gutenberg submodule hash, I think `develop` is appropriate.

To test: 

1. Verify that CI builds
2. Head over to https://github.com/wordpress-mobile/WordPress-Android/pull/14215 which uses the binary published from here and:
  - Verify it builds in CI
  - Verify you can run the app and use the editor locally, making sure the `localGutenbergMobile` path in your `local-builds.gradle` is commented out, otherwise it'll kick off a composite build, which won't use the binary.

---

PR submission checklist:

- [x] I have considered adding unit tests where possible. (N.A.)
- [x] I have considered if this change warrants user-facing release notes [more info](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/Release-notes.md) and have added them to [RELEASE-NOTES.txt](https://github.com/wordpress-mobile/gutenberg-mobile/blob/develop/RELEASE-NOTES.txt) if necessary. (N.A.)
